client/ayon_deadline/repository/custom/plugins/GlobalJobPreLoad.py

```diff
@@ -651,6 +574,142 @@ def inject_ayon_environment(deadlinePlugin):
         raise
 
 
+def _wait_for_in_progress(export_url):
+    """Check if another worker started extractenvironments already.
+
+    extractenvironments might be expensive operation, so first worker who
+    starts doing it creates empty file before it. All other workers should
+    wait until this file gets renamed to final name.
+    Raises:
+        (RuntimeError) if extraction takes more
+            than EXTRACT_ENVIRONMENT_TIMEOUT seconds
+    """
+    start_time = datetime.now()
+    while os.path.exists(f"{export_url}.tmp"):
+        date_diff = datetime.now() - start_time
+        if date_diff > timedelta(seconds=EXTRACT_ENVIRONMENT_TIMEOUT):
+            raise RuntimeError(
```

Comment:
NOTE: This does crash, si if the file is there (e.g. because process handling it failed), there there is no mechanism how to force to do it again. To unblock that case means that someone has to manually remove the file.

Reply:
> si if the file is there (e.g. because process handling it failed), there there is no mechanism how to force to do it again. To unblock that case means that someone has to manually remove the file.

Yup - that's very dangerous I'd say in production. It might be better to check for file existence, and compare datetime against it's creation time? (or maybe store some timestamp in the file?). At least something we can accurately check against whether it's not old?

Reply:
Changed to file modified date.